apps/namadillo/src/hooks/useValidateApiUrl.ts

```diff
@@ -0,0 +1,25 @@
+import { isUrlValid } from "@namada/utils";
+import { isApiAlive } from "atoms/settings";
+import { useMemo } from "react";
+
+export const useValidateApiUrl = (): ((url: string) => Promise<string>) => {
+  return useMemo(() => {
+    return async (url) => {
+      const trimmedUrl = url.trim();
```

Comment:
Because we're now reusing this, maybe it would be good to separate the concerns and move the sanitize url lines into a new function. Also, maybe `useValidateApiUrl` should return a boolean or another type rather than a string (it's a bit unexpected that this hooks returns the sanitized string, for example)

Reply:
I understood your concern, but we need to return the string because we trim and remove the last slash.
Maybe this is a naming issue. I'll propose something different soon

Reply:
I moved the logic to the write function of the atom and created the helper `changeSettingsUrl` to handle both cases. With this change, we don't have the hook for validation anymore. What do you think?

Reply:
moved this logic to `sanitizeUrl`